COMPILING.md

```diff
@@ -255,6 +255,10 @@ require manual modification of build files.
    to generate IDE projects by supplying the `-G` flag.  Run `cmake -G` for a
    comprehensive list of supported back-ends.
 
+   Note that if you do not specify the `build` directory the current directory
+   will be the build target. This will modify many version controlled files
+   and be messy to fix.
```

Comment:
I think I'll just add a change to cmake to make this impossible.

Reply:
Handled by #5782